### DISCUSSION TITLE
How to use it as library?

### DISCUSSION THREAD

**Discussion by rawhuul** [GitoxideLabs/gitoxide#1020] [Q&A] [answered]
Hey, there I want to use gitoxide in my project minimally where my aim is just to shallow clone a repository, please help me figure out.

**Comment by Byron** [comment, marked as answer]
You can take the `gix clone` (CLI) [as reference](https://github.com/Byron/gitoxide/blob/072ee32f693a31161cd6a843da6582d13efbb20b/gitoxide-core/src/repository/clone.rs#L75-L84).
